knnga_util.py

```diff
@@ -0,0 +1,593 @@
+# Copyright (C) 2020 Juliette Regimbal
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import unicode_literals
+from gamera import knnga
+
+import json
+
+
+class SerializableSelection():
+    """
+    Extension of gamera.knnga.GASelection that provides
+    access to the selection and parameters used.
+    """
+
+    def __init__(self):
+        self.method = None
+        self.parameters = {}
+        self.selection = knnga.GASelection()
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            a = self.__dict__.copy()
+            del a['selection']
+            b = self.__dict__.copy()
+            del b['selection']
+            return a == b
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def setRandomSelection(self):
+        self.method = "random"
+        self.parameters = {}
+        self.selection.setRandomSelection()
+
+    def setRankSelection(self, pressure=2.0, exponent=1.0):
```

Comment:
Magic numbers here? Maybe make them into a constant at the beginning, specially since you use the ``pressure`` more than once

Reply:
All the defaults here are the defaults set in Gamera as well. They're just as arbitrary there unfortunately, but I'll add a comment explaining that the defaults are taken from Gamera.

Reply:
@blueNoteb5 Thank you for noting the magic numbers! We should avoid them. 
@JRegimbal so do define them as constants at the beginning for now. Eventually we should create a configuration file to store all of these numbers that can be changed by the user.

Reply:
Since these numbers are the default numbers for Biollante in Gamera we should leave these values here. If the user wants to change them, they can do it in the UI of the interactive job.

Reply:
If the user can already change these, that's great.
At some point, I still think it'd be better to name these magic numbers with meaningful constant names, e.g., DEFAULT_PRESSURE and DEFAULT_EXPONENT.